go/libraries/doltcore/dbfactory/file.go

```diff
@@ -52,19 +52,50 @@ func (fact FileFactory) CreateDB(ctx context.Context, nbf *types.NomsBinFormat,
 	path = filepath.FromSlash(path)
 	path = urlObj.Host + path
 
-	info, err := os.Stat(path)
+	err = validateDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	newGenSt, err := nbs.NewLocalStore(ctx, nbf.VersionString(), path, defaultMemTableSize)
 
 	if err != nil {
 		return nil, err
-	} else if !info.IsDir() {
-		return nil, filesys.ErrIsFile
 	}
 
-	st, err := nbs.NewLocalStore(ctx, nbf.VersionString(), path, defaultMemTableSize)
+	oldgenPath := filepath.Join(path, "oldgen")
+	err = validateDir(oldgenPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
```

Comment:
I think

```suggestion
		if !errors.Is(err, os.ErrNotExist) {
```

is slightly preferred.